packages/magento-customer/hooks/useCachePersistor.ts

```diff
@@ -0,0 +1,16 @@
+import { useApolloClient } from '@graphcommerce/graphql'
+import { CachePersistor } from 'apollo-cache-persist'
+import { PersistedData, PersistentStorage } from 'apollo-cache-persist/types'
+import { useEffect, useRef } from 'react'
+
+export function useCachePersistor() {
+  const client = useApolloClient()
+  const persistor = new CachePersistor({
+    cache: client.cache,
+    storage: window.localStorage as PersistentStorage<PersistedData<string>>,
+  })
+
+  return {
+    persistor,
+  }
```

Comment:
Can you wrap this in a useMemo so it doesn't recreate the CachePersistor on each reander?